components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2006,6 +2017,24 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         throw new ResponseError(ErrorCodes.EE_FEATURE, `Admin support is implemented in Gitpod's Enterprise Edition`);
     }
 
+    async adminGetSettings(ctx: TraceContext): Promise<InstallationAdminSettings> {
+        traceAPIParams(ctx, {});
+
+        await this.guardAdminAccess("adminGetSettings", {}, Permission.ADMIN_API);
+
+        const settings = await this.installationAdminDb.getData();
+
+        return settings.settings;
+    }
+
+    async adminUpdateSettings(ctx: TraceContext, settings: InstallationAdminSettings): Promise<void> {
+        traceAPIParams(ctx, {});
+
+        await this.guardAdminAccess("adminSettingsGetTelemetry", {}, Permission.ADMIN_API);
+
+        await this.installationAdminDb.setSettings(settings);
```

Comment:
Instead of calling this directly, we need to check the properties we expect in `settings` explicitly:
```
        const newSettings: Partial<Settings> = {};
        for (const p of InstallationAdminSettings.fields()) {
            if (p in settings) {
                (newSettings[p] as any) = settings[p];
            }
        }
       await this.installationAdminDb.setSettings(newSettings);
```